src/diffusers/modular_pipelines/modular_pipeline.py

```diff
@@ -344,6 +259,23 @@ def expected_components(self) -> List[ComponentSpec]:
     def expected_configs(self) -> List[ConfigSpec]:
         return []
 
+    @property
+    def inputs(self) -> List[InputParam]:
+        """List of input parameters. Must be implemented by subclasses."""
+        return []
+
+    @property
+    def intermediate_outputs(self) -> List[OutputParam]:
+        """List of intermediate output parameters. Must be implemented by subclasses."""
+        return []
+
+    def _get_outputs(self):
+        return self.intermediate_outputs
+
+    @property
+    def outputs(self) -> List[OutputParam]:
+        return self._get_outputs()
```

Comment:
Think these can also be consolidated into just `outputs`? Didn't do it here to keep PR scope limited.

Reply:
yes to both:
we should only have `outputs`
can do it in a next PR: currently `outputs` are not used actually, so just have to remove `outputs` from the code basee and then change the `intermediate_outputs` to `outputs`